MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -2675,8 +2675,20 @@ public void readNews() {
     public int getDeploymentDeficit(AtBContract contract) {
         int total = -contract.getRequiredLances();
         int role = -Math.max(1, contract.getRequiredLances() / 2);
+        boolean hasScenarios = contract.getScenarios().size() > 0;
+
         for (Lance l : lances.values()) {
-            if (l.getMissionId() == contract.getId() && l.getRole() != Lance.ROLE_UNASSIGNED) {
+            if (hasScenarios) {
+                if (l.getMissionId() == contract.getId() && l.getRole() != Lance.ROLE_UNASSIGNED) {
+                    total++;
+                    if (l.getRole() == contract.getRequiredLanceType()) {
+                        role++;
+                    }
+                }
+            } else if (l.getMissionId() == Lance.NO_MISSION && l.getRole() != Lance.ROLE_UNASSIGNED) {
```

Comment:
I'm not sure I understand the purpose of this addition. getMissionID checks the *contract* to which a lance is assigned, rather than the scenario. A lance not assigned to a contract shouldn't be getting counted for this purpose.

Reply:
Apparently you are only assigned to a contract when you get assigned to a scenario. I can look into cleaning that up.